src/models/v3/asyncapi.ts

```diff
@@ -30,10 +32,172 @@ import type { SchemasInterface } from '../schemas';
 import type { OperationInterface } from '../operation';
 import type { ChannelInterface } from '../channel';
 import type { ServerInterface } from '../server';
+import type { BindingsInterface } from 'models/bindings';
+import type { ChannelParametersInterface } from 'models/channel-parameters';
+import type { CorrelationIdsInterface } from 'models/correlation-ids';
+import type { MessageTraitsInterface } from 'models/message-traits';
+import type { OperationTraitsInterface } from 'models/operation-traits';
+import type { ServerVariablesInterface } from 'models/server-variables';
+import type { TagsInterface } from 'models/tags';
+import type { Tag } from './tag';
+import type { TagInterface } from 'models/tag';
 
 import type { v3 } from '../../spec-types';
 
 export class AsyncAPIDocument extends BaseModel<v3.AsyncAPIObject> implements AsyncAPIDocumentInterface {
+  allSecuritySchemes(): SecuritySchemesInterface {
+    return this.securitySchemes();
```

Comment:
I see two issues here:

1. If `allSecuritySchemes` does the same as `securitySchemes`, then this is redundant.
2. I believe `securitySchemes` method implementation is wrong because it is only returning Security Schemes from `components` but instead would need to return from, for example, Operations in the root document.

cc @jonaslagoni please double 👀 just in case I'm delirious.

Reply:
> 1. If `allSecuritySchemes` does the same as `securitySchemes`, then this is redundant.

I agree. it has only been added for the sake of completeness.

> 2. I believe `securitySchemes` method [implementation](https://github.com/asyncapi/parser-js/blob/1ac0843347318c3095ae0872c55dd6a80dd44415/src/models/v3/asyncapi.ts#L260-L266) is wrong ATM because it is only returning Security Schemes from `components` but instead would need to return from, for example, Operations in the root document.

the problem is that when I try to access the security scheme of an specific `operation` or `operationTrait` it returns `SecurityRequirements[]` instead of `SecuritySchemes`. 

not sure if this is a bug or I am missing something.

Reply:
@smoya pingy pongy.